imap/mbexamine.c

```diff
@@ -351,19 +368,22 @@ static int do_examine(struct findall_data *data, void *rock __attribute__((unuse
 
     mailbox_close(&mailbox);
 
+    if (!r && ok_count) (*ok_count) ++;
```

Comment:
I think that NULL pointer check on ok_count can be omitted. Neither callback is called with a NULL rock, so better assert(ok_count) at the start of the function?

Reply:
This is just a good habit -- if your function takes an output pointer, safely allow it to be NULL in case the caller doesn't care.

We have loads of instances where over time we've re-used some prexisting callback in some slightly new context.  Will the same thing happen here?  Probably not.  But if it does, the callback already provides a sane interface, even though its current usage doesn't require it.

An assert would provide the opposite behaviour: mandating the caller either care, or modify the function (and potentially all the other callers) into this shape if they don't.

This isn't a hot path, so the sane interface and good discipline is worth more than the one instruction we'd save by removing it.

(And if we replaced it with an assert, that would be replacing a trivial compare-to-zero with an expensive branch and possible function call, because our asserts are never compiled out!)

Reply:
thanks for the explanation, I agree

Reply:
Here `r` is always `0`, so `if (!r && …)` is tautology.  Likewise `return r;` on the next line and at the end of `do_quota()`.